components/throttle.js

```diff
@@ -0,0 +1,93 @@
+// throttle.js
+
+var _ = require('underscore');
+var fs = require('fs');
+
+var idIndex = 0;
+var ids;
+
+var logger = require('../src/logger.js');
+var wrapper = require('../src/split-wrapper.js');
+
+/** 
+ * Reads a file with one record per line, and outputs a hash with the 
+ * throttle_size number of entries for processing downstream. For example, 
+ * given a file with values 1, 2, 3, 4 and a throttle size of 2, this 
+ * component would first output a hash that reads {"1": "1", "2": "2"}.  If a 
+ * subsequent throttle size packet of 1 comes in, it would then send 
+ * {"3": "3"}.  One more packet of size 1 would send {"4": "4"}.
```

Comment:
Comment is correct. 

If throttle-size is two, it will send the next two elements. Since we are just starting, it will send {"1":"1", "2":"2"}.  If it stays two, when it fires again, it would send {"3":'3", "4":"4"}.  The next elements to send are wherever it left off.